# **Project Name : CLI based Student Performance Analyzer**

# Totally Based on NumPy

In [1]:
import numpy as np

# 7 students, Marks of 5 subjects
marks = np.array([[78, 85, 92, 88, 90],   # Student 1
                  [45, 52, 38, 60, 62],   # Student 2
                  [91, 95, 88, 97, 85],   # Student 3
                  [33, 41, 29, 55, 60],   # Student 4
                  [67, 73, 81, 70, 75],   # Student 5
                  [45, 50, 55, 43, 63],   # Student 6
                  [82, 80, 95, 90, 88]])  # Student 7


## Average of Every Student :

In [ ]:
print("Average of Every Student :")
avg_marks = (np.mean(marks, axis=1))
avg_marks = np.reshape(avg_marks, (7,1))
print(avg_marks)

Average of Every Student :
[[86.6]
 [51.4]
 [91.2]
 [43.6]
 [73.2]
 [51.2]
 [87. ]]


## Average of Every Subject :

In [3]:
print("Average of Every Subject :")
avg_sub = np.mean(marks, axis=0)
print(avg_sub)


Average of Every Subject :
[63.         68.         68.28571429 71.85714286 74.71428571]


## Print Result Pass or Fail every Subject

In [4]:
result = np.where(avg_marks >=50, "Pass", "Fail")
result = np.reshape(result, (7,1))
print(result)

[['Pass']
 ['Pass']
 ['Pass']
 ['Fail']
 ['Pass']
 ['Pass']
 ['Pass']]


## Assigning Grades on the basis of Avg Marks

In [5]:

grads = np.where(avg_marks >= 90, "A", np.where(avg_marks > 75, "B", np.where(avg_marks >60, "C", np.where(avg_marks > 50, "D", "Fail"))))

print(grads)


[['B']
 ['D']
 ['A']
 ['Fail']
 ['C']
 ['D']
 ['B']]


## Top Performer

In [6]:
topper = np.argmax(avg_marks)
print(f"Topper of the class is : Student {topper+1}")


Topper of the class is : Student 3


## Toughest Subject

In [7]:
tough_sub = np.argmin(avg_sub)
print(f"Toughest subject of the class is Subject {tough_sub+1}")


Toughest subject of the class is Subject 1


## Flagging subject as Low Marks

In [8]:
low_marks = np.where(marks < 40, "Low Marks", "OK")

print(low_marks)


[['OK' 'OK' 'OK' 'OK' 'OK']
 ['OK' 'OK' 'Low Marks' 'OK' 'OK']
 ['OK' 'OK' 'OK' 'OK' 'OK']
 ['Low Marks' 'OK' 'Low Marks' 'OK' 'OK']
 ['OK' 'OK' 'OK' 'OK' 'OK']
 ['OK' 'OK' 'OK' 'OK' 'OK']
 ['OK' 'OK' 'OK' 'OK' 'OK']]


## Observations:

- Student 2 has low marks in Subject 3 (38) — needs improvement
- Student 4 has low marks in Subject 1 (33) and Subject 3 (29) — at risk
- Student 3 is the top performer with 91.2 average — Grade A
- Subject 1 is the toughest with lowest class average (63.0)